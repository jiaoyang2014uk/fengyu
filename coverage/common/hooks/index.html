
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for common/hooks</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> common/hooks</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>266/266</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">96.42% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>54/56</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">90% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>18/20</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>266/266</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="useClass.ts"><a href="useClass.ts.html">useClass.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="27" class="abs high">27/27</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="27" class="abs high">27/27</td>
	</tr>

<tr>
	<td class="file high" data-value="useColumns.ts"><a href="useColumns.ts.html">useColumns.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="33" class="abs high">33/33</td>
	<td data-value="66.66" class="pct medium">66.66%</td>
	<td data-value="6" class="abs medium">4/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="33" class="abs high">33/33</td>
	</tr>

<tr>
	<td class="file high" data-value="useFormattedData.ts"><a href="useFormattedData.ts.html">useFormattedData.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="105" class="abs high">105/105</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="8" class="abs medium">6/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="105" class="abs high">105/105</td>
	</tr>

<tr>
	<td class="file high" data-value="usePagination.ts"><a href="usePagination.ts.html">usePagination.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="37" class="abs high">37/37</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="37" class="abs high">37/37</td>
	</tr>

<tr>
	<td class="file high" data-value="useSortData.ts"><a href="useSortData.ts.html">useSortData.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="31" class="abs high">31/31</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="31" class="abs high">31/31</td>
	</tr>

<tr>
	<td class="file high" data-value="useSortable.ts"><a href="useSortable.ts.html">useSortable.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="33" class="abs high">33/33</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="33" class="abs high">33/33</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Wed May 25 2022 00:59:48 GMT+0800 (China Standard Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    