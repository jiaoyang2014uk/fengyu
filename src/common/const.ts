/**
 * 常量
 */

import {PropType} from "vue";
import type {PaginationOptions, ColumnOptions, SortOptions, TableData,} from './types'

// 定义排序顺序常量
export enum SORT_ORDER {
    /* 无排序 */
    NONE = 'NONE',
    /* 升序排序 */
    ASC = 'ASC',
    /* 降序排序 */
    DESC = 'DESC'
}

//   定义排序样式常量
export const sortClsMap = {
    [SORT_ORDER.NONE]: 'table__header-cell_sort',
    [SORT_ORDER.ASC]: 'table__header-cell_sort_asc',
    [SORT_ORDER.DESC]: 'table__header-cell_sort_desc',
};
//   定义下一次排序顺序
export const nextSortOrderMap = {
    [SORT_ORDER.NONE]: SORT_ORDER.ASC,
    [SORT_ORDER.ASC]: SORT_ORDER.DESC,
    [SORT_ORDER.DESC]: SORT_ORDER.NONE
};

//  定义 SimpleTable 的props常量
export const tableProps = {
    /* 表格列配置 */
    columns: {
        type: Array as PropType<ColumnOptions[]>,
        default: () => [],
    },
    /* 显示数据配置 */
    data: {
        type: Array as PropType<TableData[]>,
        default: () => [],
    },
    /* 排序配置 */
    sortOptions: {
        type: Object as PropType<SortOptions>,
        default: () => ({})
    },
    paginationOptions: {
        type: Object as PropType<PaginationOptions>,
        default: () => ({})
    },
} as const;

//   定义 TableCol 的props常量
export const tableColProps = {
    /* 表格列配置 */
    columns: {
        type: Array as PropType<ColumnOptions[]>,
        default: () => [],
    }
} as const;

//   定义 TableHeader 的props常量
export const tableHeaderProps = {
    /* 表格列配置 */
    columns: {
        type: Array as PropType<ColumnOptions[]>,
        default: () => [],
    },
    /* 排序配置 */
    sortOptions: {
        type: Object as PropType<SortOptions>,
        default: () => ({})
    },
    /* 本地排序方法 */
    onChangeSort: {
        type: Function,
        default: () => {
        },
    }
} as const;

//   定义 TableHeaderRow 的props常量
export const tableHeaderRowProps = {
    /* 表格列配置 */
    columns: {
        type: Array as PropType<ColumnOptions[]>,
        default: () => [],
    },
    /* 排序配置 */
    sortOptions: {
        type: Object as PropType<SortOptions>,
        default: () => ({})
    },
    /* 本地排序方法 */
    onChangeSort: {
        type: Function,
        default: () => {
        },
    }
} as const;

//   定义 TableHeaderCell 的props常量
export const tableHeaderCellProps = {
    /* 表格列配置 */
    column: {
        type: Object as PropType<ColumnOptions>,
        default: () => [],
    },
    /* 排序配置 */
    sortOptions: {
        type: Object as PropType<SortOptions>,
        default: () => ({})
    },
    /* 本地排序方法 */
    onChangeSort: {
        type: Function,
        default: () => {
        },
    }
} as const;

//   定义 TableHeaderSort 的props常量
export const tableHeaderSortProps = {
    /* 表格列配置 */
    column: {
        type: Object as PropType<ColumnOptions>,
        default: () => [],
    },
    /* 排序配置 */
    sortOptions: {
        type: Object as PropType<SortOptions>,
        default: () => ({})
    },
    /* 本地排序方法 */
    onChangeSort: {
        type: Function,
        default: () => {
        },
    }
} as const;

//   定义 TableBody 的props常量
export const tableBodyProps = {
    /* 表格列配置 */
    columns: {
        type: Array as PropType<ColumnOptions[]>,
        default: () => [],
    },
    /* 显示数据配置 */
    data: {
        type: Array as PropType<TableData[]>,
        default: () => [],
    },
} as const;

//   定义 TableBodyRow 的props常量
export const tableBodyRowProps = {
    /* 表格列配置 */
    columns: {
        type: Array as PropType<ColumnOptions[]>,
        default: () => [],
    },
    /* 显示数据配置 */
    data: {
        type: Array as PropType<TableData[]>,
        default: () => [],
    },
    /* 每行数据 */
    record: {
        type: Object as PropType<TableData>,
        default: () => [],
    },
} as const;

//   定义 TableBodyCell 的props常量
export const tableBodyCellProps = {
    /* 表格列配置 */
    column: {
        type: Object as PropType<ColumnOptions>,
        default: () => ({}),
    },
    /* 显示数据配置 */
    data: {
        type: Object as PropType<TableData>,
        default: () => ({}),
    },
} as const;

//   定义 Pagination 的props常量
export const paginationProps = {
    /* 是否支持远端分页 */
    remote: {
        type: Boolean,
        default: false
    },
    /* 每页显示条目个数 */
    pageSize: {
        type: Number,
        default: 10
    },
    /* 页数 */
    current: {
        type: Number,
        default: 1
    },
    /* 总条目数 */
    total: {
        type: Number,
        default: 0
    },
    /* 本地分页方法 */
    onChangePage: {
        type: Function,
        default: () => {
        },
    }
} as const;
