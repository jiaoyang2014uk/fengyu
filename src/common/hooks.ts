import {computed} from "vue";
import type {ColumnOptions, PaginationOptions, Slots, SortOptions, TableData} from "./types";
import {SORT_ORDER} from "./const";

//SimpleTable hooks
/**
 * 分页总数取值
 * @param {ColumnOptions} columns 表格列配置
 * @param {Slots} slots 插槽
 * @returns {ColumnOptions} headerColumns 表格头配置
 * @returns {ColumnOptions} bodyColumns 表格列配置
 */
export function useColumnsMethod(columns: ColumnOptions[], slots: Slots) {
    // 根据名称取 slot，注入到 column 配置中
    const headerColumns = (columns?.map(column => ({
        ...column,
        slot: slots[`header__${column.dataIndex}`]
    })) || []) as ColumnOptions[];
    const bodyColumns = (columns?.map(column => ({
        ...column,
        slot: slots[`body__${column.dataIndex}`]
    })) || []) as ColumnOptions[];

    return {
        headerColumns,
        bodyColumns
    };
};

/**
 * 根据排序和分页获取格式化后的数据
 * @param {TableOptions} props 表格配置
 * @returns {TableData} formattedData 格式化后数据
 */
export function useFormattedDataMethod(props: {
    data: TableData[],
    sortOptions: SortOptions,
    columns: ColumnOptions[],
    paginationOptions: PaginationOptions,
}) {
    const sortHandler = ({data, sortOptions, columns}: {
        data: TableData[],
        sortOptions: SortOptions,
        columns: ColumnOptions[]
    }) => {
        let sortedData = [...data];
        if (sortOptions.remote || sortOptions.sortOrder === SORT_ORDER.NONE) {
            return sortedData;
        }

        const sortColumn = columns?.find(col => col.dataIndex === sortOptions.sortBy);
        let defaultFn = (a: TableData, b: TableData) => a[sortOptions.sortBy] > b[sortOptions.sortBy] ? 1 : -1;
        const sortFn = sortColumn?.sortFn || defaultFn;
        const descSortFn = (a: TableData, b: TableData) => -sortFn(a, b);

        sortedData.sort(sortOptions.sortOrder === SORT_ORDER.ASC ? sortFn : descSortFn);

        return sortedData;
    };
    const paginationHandler = ({data, paginationOptions}: {
        data: TableData[],
        paginationOptions: PaginationOptions
    }) => {
        const paginationData = [...data];
        if (paginationOptions.remote || !paginationOptions.enable) {
            return paginationData;
        }

        const start = paginationOptions.limit * (paginationOptions.page - 1);
        const end = start + paginationOptions.limit;
        return paginationData.slice(start, end);
    };
    return computed(() => {
        // 先排序再分页
        let data = sortHandler({
            data: props.data,
            sortOptions: props.sortOptions,
            columns: props.columns
        });
        data = paginationHandler({
            data: data,
            paginationOptions: props.paginationOptions
        });
        return data;
    });
}

/**
 * 分页总数取值
 * @param {PaginationOptions} paginationOptions 分页配置
 * @param {TableData[]} data 显示数据
 * @returns {number} total 条目总数
 */
export function usePaginationMethod(paginationOptions: PaginationOptions, data: TableData[]) {
    // 远端分页 total 由外部传入，本地分页 total = data.length
    return computed(() => paginationOptions.remote ? paginationOptions.total : data.length);
}

// TableHeaderCell hooks

/**
 * 排序处理方法
 * @param {TableHeaderCellProps} props TableHeaderCell 的props
 * @returns {string} sortCls 排序样式
 * @returns {string} nextOrder 下一次排序顺序
 */
export function useSortMethod(props: {
    column: ColumnOptions,
    sortOptions: SortOptions
}) {
    const sortClsMap = {
        [SORT_ORDER.NONE]: 'table__header-cell_sort',
        [SORT_ORDER.ASC]: 'table__header-cell_sort_asc',
        [SORT_ORDER.DESC]: 'table__header-cell_sort_desc',
    };
    const nextSortOrderMap = {
        [SORT_ORDER.NONE]: SORT_ORDER.ASC,
        [SORT_ORDER.ASC]: SORT_ORDER.DESC,
        [SORT_ORDER.DESC]: SORT_ORDER.NONE
    };
    const curOrder = computed(() => props.column.dataIndex === props.sortOptions.sortBy ? props.sortOptions.sortOrder : SORT_ORDER.NONE);
    const nextOrder = computed(() => nextSortOrderMap[curOrder.value]);
    const sortCls = computed(() => sortClsMap[props.column.dataIndex === props.sortOptions.sortBy ? props.sortOptions.sortOrder : SORT_ORDER.NONE]);
    return {
        sortCls,
        nextOrder
    };
}

// Pagination hooks

/**
 * 分页上一页/下一页样式
 * @param {PaginationOptions} options 分页配置
 * @param {number} totalPage 总页数
 * @returns {string} previousCls 上一页样式
 * @returns {string} nextCls 下一页样式
 */
export function useClassMethod(options: PaginationOptions, totalPage: number) {
    const previousCls = computed(() => `pagination__previous ${options.page <= 1 ? 'disabled' : ''}`);
    const nextCls = computed(() => `pagination__next ${options.page >= totalPage ? 'disabled' : ''}`);
    return {
        previousCls,
        nextCls
    };
}