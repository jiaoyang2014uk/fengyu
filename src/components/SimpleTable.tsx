import {defineComponent} from "vue";
import TableHeader from "./TableHeader";
import TableBody from "./TableBody";
import Pagination from "./Pagination";
import {useColumnsMethod, useFormattedDataMethod, usePaginationMethod} from '../common/hooks/useTable'
import {debug, trace} from "../common/logger";
import '../common/Table.less';
import type {TableProps, SortOptions} from "../common/types";
import {tableProps} from "../common/const";

export default defineComponent({
    name: "SimpleTable",
    props: tableProps,
    setup(props: TableProps, {attrs, emit, slots}) {
        const {headerColumns, bodyColumns} = useColumnsMethod(props.columns, slots);
        const formattedData = useFormattedDataMethod(props);
        const total = usePaginationMethod(props.paginationOptions, props.data);
        const onSortChange = (sortOptions: SortOptions) => {
            //todo 当前的sort状态的维护
            debug(
                '[SimpleTable]: ',
                `onSortChange emit updateSortOptions with sortOptions: ${JSON.stringify(sortOptions)}`
            )
            emit('updateSortOptions', sortOptions);
        };
        const onUpdatePage = (page: number) => {
            const options = {
                ...props.paginationOptions,
                page
            };
            debug(
                '[SimpleTable]: ',
                `onUpdatePage emit updatePaginationOptions with options: ${JSON.stringify(options)}`
            )
            emit('updatePaginationOptions', options);
        };
        return () => {
            trace(
                '[SimpleTable]: ',
                `columns: ${JSON.stringify(props.columns)}, data: ${JSON.stringify(props.data)}, sortOptions: ${JSON.stringify(props.sortOptions)}, paginationOptions: ${JSON.stringify(props.paginationOptions)}`
            )
            return (
                <div class='table'>
                    <table class="table__table">
                        <TableHeader columns={headerColumns} sortOptions={props.sortOptions}
                                     onUpdateSortOptions={onSortChange}/>
                        <TableBody columns={bodyColumns} data={formattedData.value}/>
                    </table>
                    {
                        props.paginationOptions.enable &&
                        <Pagination v-show={props.paginationOptions.enable}
                                    limit={props.paginationOptions.limit}
                                    page={props.paginationOptions.page}
                                    total={total.value}
                                    onUpdatePage={onUpdatePage}/>
                    }
                </div>
            );
        };
    },
});
