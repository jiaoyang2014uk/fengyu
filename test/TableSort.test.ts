import {mount} from "@vue/test-utils";
import {expect, test} from "vitest";
import SimpleTable from "../src/components/SimpleTable";
import type {TableData} from "../src/common/types";
import {SORT_ORDER} from "../src/common/const";

const mountTable = (options?: Record<string, unknown>) => mount(SimpleTable, options);

test('sort: build-in sort', async () => {
    let wrapper = mountTable({
        props: {
            columns: [
                {header: 'Name', dataIndex: 'name'},
                {header: 'Age', dataIndex: 'age', sortable: true}
            ],
            data: [
                {name: 'Bob', age: '27'},
                {name: 'Tom', age: '20'},
                {name: 'Alice', age: '23'}
            ],
            sortOptions: {
                sortBy: '',
                sortOrder: SORT_ORDER.NONE
            }
        },
    });

    //点击一次升序
    await wrapper.findAll('.table__header-cell_sort')[0].trigger('click')
    expect(wrapper.emitted('updateSortOptions')).toHaveLength(1)
    expect(wrapper.emitted('updateSortOptions')).toEqual([[{
        remote: undefined,
        sortBy: 'age',
        sortOrder: SORT_ORDER.ASC
    }]])
    expect(wrapper.html()).toMatchSnapshot();

    //点击一次降序
    await wrapper.findAll('.table__header-cell_sort')[0].trigger('click')
    expect(wrapper.html()).toMatchSnapshot();

    //点击一次无序
    await wrapper.findAll('.table__header-cell_sort')[0].trigger('click')
    expect(wrapper.html()).toMatchSnapshot();

    //点击未排序列 无效
    await wrapper.findAll('.table__header-cell__value')[0].trigger('click')
    expect(wrapper.html()).toMatchSnapshot();
});

test('sort: custom sort', async () => {
    const sortFn = (a: TableData, b: TableData) => a.age < b.age ? 1 : -1;
    const wrapper = mountTable({
        props: {
            columns: [
                {header: 'Name', dataIndex: 'name'},
                {header: 'Age', dataIndex: 'age', sortable: true, sortFn}
            ],
            data: [
                {name: 'Bob', age: '27'},
                {name: 'Tom', age: '20'},
                {name: 'Alice', age: '23'}
            ]
        },
    });

    //点击一次升序
    await wrapper.findAll('.table__header-cell_sort')[0].trigger('click')
    expect(wrapper.html()).toMatchSnapshot();

    //点击一次降序
    await wrapper.findAll('.table__header-cell_sort')[0].trigger('click')
    expect(wrapper.html()).toMatchSnapshot();

    //点击一次无序
    await wrapper.findAll('.table__header-cell_sort')[0].trigger('click')
    expect(wrapper.html()).toMatchSnapshot();

    //点击未排序列 无效
    await wrapper.findAll('.table__header-cell__value')[0].trigger('click')
    expect(wrapper.html()).toMatchSnapshot();
});

test('sort: remote sort', async () => {
    const wrapper = mountTable({
        props: {
            columns: [
                {header: 'Name', dataIndex: 'name'},
                {header: 'Age', dataIndex: 'age', sortable: true}
            ],
            data: [
                {name: 'Bob', age: '27'},
                {name: 'Tom', age: '20'},
                {name: 'Alice', age: '23'}
            ],
            sortOptions: {
                sortBy: '',
                sortOrder: SORT_ORDER.NONE,
                remote: true
            }
        },
    });

    // 点击无效果
    await wrapper.findAll('.table__header-cell_sort')[0].trigger('click')
    expect(wrapper.html()).not.toContain('table__header-cell_sort_asc');
    expect(wrapper.html()).toMatchSnapshot();
});

