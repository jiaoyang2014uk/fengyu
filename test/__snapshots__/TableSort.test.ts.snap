// Vitest Snapshot v1

exports[`sort: build-in sort 1`] = `
"<div class=\\"table\\">
  <table class=\\"table__table\\">
    <thead>
      <tr>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Name</span>
          <!---->
        </th>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Age</span><span class=\\"table__header-cell_sort\\"></span></th>
      </tr>
    </thead>
    <tbody>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Bob</span></td>
        <td class=\\"table__body-cell\\"><span>27</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Tom</span></td>
        <td class=\\"table__body-cell\\"><span>20</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Alice</span></td>
        <td class=\\"table__body-cell\\"><span>23</span></td>
      </tr>
    </tbody>
  </table>
  <!---->
</div>"
`;

exports[`sort: build-in sort 2`] = `
"<div class=\\"table\\">
  <table class=\\"table__table\\">
    <thead>
      <tr>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Name</span>
          <!---->
        </th>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Age</span><span class=\\"table__header-cell_sort\\"></span></th>
      </tr>
    </thead>
    <tbody>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Bob</span></td>
        <td class=\\"table__body-cell\\"><span>27</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Tom</span></td>
        <td class=\\"table__body-cell\\"><span>20</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Alice</span></td>
        <td class=\\"table__body-cell\\"><span>23</span></td>
      </tr>
    </tbody>
  </table>
  <!---->
</div>"
`;

exports[`sort: build-in sort 3`] = `
"<div class=\\"table\\">
  <table class=\\"table__table\\">
    <thead>
      <tr>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Name</span>
          <!---->
        </th>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Age</span><span class=\\"table__header-cell_sort\\"></span></th>
      </tr>
    </thead>
    <tbody>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Bob</span></td>
        <td class=\\"table__body-cell\\"><span>27</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Tom</span></td>
        <td class=\\"table__body-cell\\"><span>20</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Alice</span></td>
        <td class=\\"table__body-cell\\"><span>23</span></td>
      </tr>
    </tbody>
  </table>
  <!---->
</div>"
`;

exports[`sort: build-in sort 4`] = `
"<div class=\\"table\\">
  <table class=\\"table__table\\">
    <thead>
      <tr>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Name</span>
          <!---->
        </th>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Age</span><span class=\\"table__header-cell_sort\\"></span></th>
      </tr>
    </thead>
    <tbody>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Bob</span></td>
        <td class=\\"table__body-cell\\"><span>27</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Tom</span></td>
        <td class=\\"table__body-cell\\"><span>20</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Alice</span></td>
        <td class=\\"table__body-cell\\"><span>23</span></td>
      </tr>
    </tbody>
  </table>
  <!---->
</div>"
`;

exports[`sort: custom sort 1`] = `
"<div class=\\"table\\">
  <table class=\\"table__table\\">
    <thead>
      <tr>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Name</span>
          <!---->
        </th>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Age</span><span class=\\"table__header-cell_sort\\"></span></th>
      </tr>
    </thead>
    <tbody>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Bob</span></td>
        <td class=\\"table__body-cell\\"><span>27</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Tom</span></td>
        <td class=\\"table__body-cell\\"><span>20</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Alice</span></td>
        <td class=\\"table__body-cell\\"><span>23</span></td>
      </tr>
    </tbody>
  </table>
  <!---->
</div>"
`;

exports[`sort: custom sort 2`] = `
"<div class=\\"table\\">
  <table class=\\"table__table\\">
    <thead>
      <tr>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Name</span>
          <!---->
        </th>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Age</span><span class=\\"table__header-cell_sort\\"></span></th>
      </tr>
    </thead>
    <tbody>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Bob</span></td>
        <td class=\\"table__body-cell\\"><span>27</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Tom</span></td>
        <td class=\\"table__body-cell\\"><span>20</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Alice</span></td>
        <td class=\\"table__body-cell\\"><span>23</span></td>
      </tr>
    </tbody>
  </table>
  <!---->
</div>"
`;

exports[`sort: custom sort 3`] = `
"<div class=\\"table\\">
  <table class=\\"table__table\\">
    <thead>
      <tr>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Name</span>
          <!---->
        </th>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Age</span><span class=\\"table__header-cell_sort\\"></span></th>
      </tr>
    </thead>
    <tbody>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Bob</span></td>
        <td class=\\"table__body-cell\\"><span>27</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Tom</span></td>
        <td class=\\"table__body-cell\\"><span>20</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Alice</span></td>
        <td class=\\"table__body-cell\\"><span>23</span></td>
      </tr>
    </tbody>
  </table>
  <!---->
</div>"
`;

exports[`sort: custom sort 4`] = `
"<div class=\\"table\\">
  <table class=\\"table__table\\">
    <thead>
      <tr>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Name</span>
          <!---->
        </th>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Age</span><span class=\\"table__header-cell_sort\\"></span></th>
      </tr>
    </thead>
    <tbody>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Bob</span></td>
        <td class=\\"table__body-cell\\"><span>27</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Tom</span></td>
        <td class=\\"table__body-cell\\"><span>20</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Alice</span></td>
        <td class=\\"table__body-cell\\"><span>23</span></td>
      </tr>
    </tbody>
  </table>
  <!---->
</div>"
`;

exports[`sort: remote sort 1`] = `
"<div class=\\"table\\">
  <table class=\\"table__table\\">
    <thead>
      <tr>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Name</span>
          <!---->
        </th>
        <th class=\\"table__header-cell\\"><span class=\\"table__header-cell__value\\">Age</span><span class=\\"table__header-cell_sort\\"></span></th>
      </tr>
    </thead>
    <tbody>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Bob</span></td>
        <td class=\\"table__body-cell\\"><span>27</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Tom</span></td>
        <td class=\\"table__body-cell\\"><span>20</span></td>
      </tr>
      <tr class=\\"table__body-row\\">
        <td class=\\"table__body-cell\\"><span>Alice</span></td>
        <td class=\\"table__body-cell\\"><span>23</span></td>
      </tr>
    </tbody>
  </table>
  <!---->
</div>"
`;
